cpp/src/arrow/memory_pool.h

```diff
@@ -87,6 +87,11 @@ class ARROW_EXPORT MemoryPool {
   ///   faster deallocation if supported by its backend.
   virtual void Free(uint8_t* buffer, int64_t size) = 0;
 
+  /// Returns unused memory to the OS
+  ///
+  /// Only applies to allocators that hold onto unused memory
```

Comment:
Mention that this is a best effort only? Even an allocator which exposes such a facility may not release all "unused" memory (for example because of memory fragmentation).

Reply:
Done.